daemon/containerd/image_history.go

```diff
@@ -103,26 +103,28 @@ func (i *ImageService) ImageHistory(ctx context.Context, name string) ([]*imaget
 		return imgs
 	}
 
+	is := i.client.ImageService()
 	currentImg := img
 	for _, h := range history {
-		h.ID = currentImg.Target.Digest.String()
-		imgs := findParents(currentImg)
+		dgst := currentImg.Target.Digest
+		h.ID = dgst.String()
+
+		imgs, err := is.List(ctx, "target.digest=="+dgst.String())
```

Comment:
nit: avoid doing the `.String()` twice since you don't need `currentImg.Target.Digest` for anything else
```suggestion
		dgst := currentImg.Target.Digest.String()
		h.ID = dgst

		imgs, err := is.List(ctx, "target.digest=="+dgst)
```